Fix: deprecated strftime with IntlDateFormatter

rmeister opened the pull request:
A little band-aid for the deprecation of strftime.

rmeister committed:
Fix: deprecated strftime with IntlDateFormatter

mzur commented:
Thanks! Is this intentionally still a draft?

mzur commented on snippets/calendar-div.php:
```diff
@@ -1,6 +1,9 @@
 <?php
 	$tmpDate = getdate(0);
 	$currentDate = getdate();
+  if (!empty($languageCode)) {
```
This should probably be:
```suggestion
  if (!isset($languageCode)) {
```

right?

mzur commented on snippets/calendar-table.php:
```diff
@@ -1,6 +1,9 @@
 <?php
 	$tmpDate = getdate(0);
 	$currentDate = getdate();
+  if (!empty($languageCode)) {
```
```suggestion
  if (!isset($languageCode)) {
```

mzur commented on snippets/calendar-teaser.php:
```diff
@@ -1,12 +1,15 @@
 <ul class="teaser">
 <?php
+  if (!empty($languageCode)) {
```
```suggestion
  if (!isset($languageCode)) {
```

mzur reviewed (commented):
Thanks a lot! Do you use this already on a site somewhere (and everything works)?

I noticed that some changes replaced the old tab indentation with spaces. Could you please change the indentation back to tabs?

rmeister committed:
Use isset instead of empty, reformat spaces to tabs, adds doc.

rmeister commented:
I use it on https://hog-neppendorf.de
It seems there still is an issue that could be related to timezones, as begin dates are incorrectly printed as one day before the specified day.

rmeister committed:
Fix timezone issues by using the DateTime ctor instead of strtotime()

The date and time fields of the data structure are assumed to be stored including the respective
 timezone. Hence, when creating the Unix timestamp from the fields the timezone must be respected,
 so the IntlDateFormatter can format the string correctly.
Instead of strtotime() it is now using the DateTime constructor with a timezone
 (from date_default_timezone_get()) as argument.
Alternatively, one could store the beginTimestamp directly as DateTime(Immutable) object.

rmeister committed:
Timestamp members of Event use DateTimeImmutable instead of Unix time integers.

Getters still give the Unix timestamp for backward compatibility.

rmeister committed:
Fix: creation of DateTimeImmutable

The right timezone seems to be already used implicitly, so it must not be applied additionally.

rmeister committed:
Rename private function getTimestamp -> getDate, and fix docs.

mzur commented:
@rmeister Does everything work for you now?

rmeister commented:
Yes, looks good so far :)

mzur reviewed (approved):
Thanks!
